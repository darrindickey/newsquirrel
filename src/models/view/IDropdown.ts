import { IDropdownItem } from './IDropdown';

export interface IDropdownItem {
  name: string;
  code: string;
  onSelect: (name: string, code: string) => void;
  icon: string;
}

export interface IDropdown {
  selectedItem: string;
  items: IDropdownItem[];
  onClick: () => void;
  onSelect: (name: string, code: string) => void;
  show: boolean;
  label: string;
  update: (data: any) => void;
}