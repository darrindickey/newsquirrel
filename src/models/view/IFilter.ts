import { IFilter } from '../data/IFilter';

export interface IFilter {
  name: string;
  value: string;
  selectFilter: (filter: IFilter) => void;
  selected: boolean;
};