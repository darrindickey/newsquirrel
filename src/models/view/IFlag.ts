export interface IFlag {
  code: string;
}