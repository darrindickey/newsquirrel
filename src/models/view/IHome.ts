import ReadingMode from "../../enums/readingMode";

export interface IHome {
  mode: ReadingMode
}