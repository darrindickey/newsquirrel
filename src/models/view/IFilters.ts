import { IFilter } from './IFilter';

export interface IFilters {
  items: IFilter[];
  update: (data: any) => void;
}