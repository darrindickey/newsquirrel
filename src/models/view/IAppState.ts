import { AppMode } from './../../enums/appMode';

import NewsStandSize from '../../enums/newsStandSize';
import { IDateFilter } from './../data/IDateFilter';
import { IArticleCard } from './IArticleCard';


interface IThumbnail {
  media: {
    height: number;
    width: number;
  },
  href: string;
}

interface ILogo {
  media: {
    height: number;
    width: number;
  },
  href: string;
}

interface ILinks {
  thumbnail: IThumbnail[];
  logo: ILogo[];
}

export interface IFrameData {
  url: string;
  meta: {
    date: string;
    description: string;
    site: string;
    author: string;
    title: string;
  };
  links: ILinks;
}

export interface INewsState {
  topHeadlines: IArticleCard[];
  newsArticles: IArticleCard[];
  totalResults: number;
  selectedCountry: string;
  searchResultsCount: number;
  headlinesCount: number;
  isAppBusy: boolean;
  sortByTime: ISortBy[];
  activeSortByTime: ISortBy;
  quickViewEnabled: boolean;
  quickViewData: IFrameData | {};
  quickViewUrl: string;
  quickViewLoading: boolean;
  quickViewPosition: {
    x: number;
    y: number;
  }
  pocketConfig: IPocketState;
  searchTerm: string;
}

export interface ICountry {
  name: string;
  value: string;
}

export interface ISortBy {
  name: string;
  value: string;
}

export interface IAppMode{
  name: string;
  value: AppMode;
}

export interface IOptionsState {
  pageSize: number;
  activePage: number;
  defaultCountries: ICountry[];
  defaultCategories: string[];
  filter: {
    categories: string[]
  },
  sortBy: ISortBy[];
  currentlySortingBy: ISortBy;
  searchingFor: string;
  newsStandSize: NewsStandSize;
  dateFilter: IDateFilter;
  detailedPaneOpen: boolean;
  detailedArticleUrl: string;
  defaultAppMode: IAppMode;
  availableAppModes: IAppMode[];
}

export interface IPocketState {
  requestToken: string;
  accessToken: string;
}

export interface IAppState {
  categories: string[];
  news: INewsState;
  options: IOptionsState
}
