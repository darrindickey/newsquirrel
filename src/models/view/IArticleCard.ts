import NewsStandSize from "../../enums/newsStandSize";
import { IAppMode } from './IAppState';

export interface IArticleCard {
  id: string;
  author: string;
  title: string;
  url: string;
  urlToImage: string;
  description: string;
  publishedAt: string;
  source: string;
}

export interface IArticleCardView extends IArticleCard{
  size: NewsStandSize;
  newSize: NewsStandSize;
  imageLoaded: boolean;
  onImageLoaded: () => void;
  showArticle: (url: string) => void;
  checkArticle: (url: string) => void;
  canEmbedInFrame: boolean;
  appMode: IAppMode;
  updateQuickviewUrl: (data: any) => void;
}