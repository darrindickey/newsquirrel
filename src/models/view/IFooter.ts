import { IAppMode } from './IAppState';

export default interface IFooter {
  appMode: IAppMode,
  mutate: any
}