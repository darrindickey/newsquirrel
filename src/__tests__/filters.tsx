import 'jest-dom/extend-expect';
import * as React from 'react';
import {cleanup, fireEvent, render } from 'react-testing-library';
import Filters from '../components/filters/filters';
import { IFilter } from '../models/view/IFilter';

afterEach(cleanup);

test('Renders Tabs correctly', () => {
  const callback = jest.fn();
  const filters: IFilter[] = [
    {
      name: 'Tab One',
      selected: false,
      value: 'Tab One',
    },{
      name: 'Tab Two',
      selected: false,
      value: 'Tab Two',
    },{
      name: 'Tab Three',
      selected: true,
      value: 'Tab Three',
    }]
  const {getByTestId} = render(<Filters items={filters} selectFilter={callback} />);
  const node = getByTestId('main-tabs');
  const list = node.querySelector('ul');
  const listItems = list!.querySelectorAll('li');

  expect(node).toBeInTheDocument();
  expect(node).toHaveClass('tabs');
  expect(listItems).toHaveLength(3);
  expect(node).toMatchSnapshot();
  fireEvent(listItems[0], new MouseEvent('click', {
    bubbles: true,
    cancelable: true
  }));
  expect(callback).toHaveBeenCalledTimes(1);
})
