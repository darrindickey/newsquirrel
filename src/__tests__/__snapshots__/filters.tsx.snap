// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Tabs correctly 1`] = `
<div
  class="tabs is-medium sc-bwzfXH cxdjHF"
  data-testid="main-tabs"
  style=""
>
  <ul
    class="sc-bdVaJa jtwmQU"
  >
    <li
      class="sc-htpNat liJpkY"
    >
      <a
        href="javascript:void(0);"
        role="button"
      >
        Tab One
      </a>
    </li>
    <li
      class="sc-htpNat liJpkY"
    >
      <a
        href="javascript:void(0);"
        role="button"
      >
        Tab Two
      </a>
    </li>
    <li
      class="selected is-active sc-htpNat eAdLHf"
    >
      <a
        href="javascript:void(0);"
        role="button"
      >
        Tab Three
      </a>
    </li>
  </ul>
</div>
`;
